include/variables/FoamVariableField.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "Moose.h"
+#include "MooseTypes.h"
+#include "FoamMesh.h"
+
+class FoamVariableField : public MooseObject
```

Comment:
Would `FoamFieldVariable` be a better name here? It's a MOOSE variable representing an OpenFOAM field, so it feels like "foam" and "field" belong next to one-another.